src/librustc_hir/def_id.rs

```diff
@@ -86,8 +87,16 @@ impl fmt::Display for CrateNum {
     }
 }
 
-impl rustc_serialize::UseSpecializedEncodable for CrateNum {}
-impl rustc_serialize::UseSpecializedDecodable for CrateNum {}
+impl rustc_serialize::UseSpecializedEncodable for CrateNum {
```

Comment:
Since we decided to stick with this approach (https://github.com/rust-lang/rust/pull/67516#issuecomment-576695916), we may want to add a comment (here or on `CrateNum`) saying which conditions have to hold when using this.